test/testkit/mocks/stdoutloggen/stdoutloggen.go

```diff
@@ -0,0 +1,101 @@
+package stdoutloggen
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	kitk8s "github.com/kyma-project/telemetry-manager/test/testkit/k8s"
+)
+
+const (
+	DefaultName          = "stdoutloggen"
+	DefaultContainerName = "stdoutloggen"
+	DefaultImageName     = "europe-docker.pkg.dev/kyma-project/dev/stdout-log-generator:PR-2357"
+)
+
+type Option func(*corev1.PodSpec)
+
+// WithContainer sets the name of the container in the pod spec
+func WithContainer(container string) Option {
+	return func(spec *corev1.PodSpec) {
+		spec.Containers[0].Name = container
+	}
+}
+
+// WithRate sets approximately how many logs per second each worker should generate. Zero means no throttling
+func WithRate(rate int) Option {
+	return func(spec *corev1.PodSpec) {
+		// find the rate argument and replace it
+		for i, arg := range spec.Containers[0].Args {
+			if arg == "--rate" {
+				spec.Containers[0].Args[i+1] = fmt.Sprintf("%d", rate)
+				return
+			}
+		}
+	}
+}
+
+// WithField sets a field in logs generated by the stdout log generator
+func WithFields(fields map[string]string) Option {
+	return func(spec *corev1.PodSpec) {
+		for k, v := range fields {
+			spec.Containers[0].Args = append(spec.Containers[0].Args, "--fields")
+			spec.Containers[0].Args = append(spec.Containers[0].Args, fmt.Sprintf("%s=%s", k, v))
+		}
+	}
+}
+
+// WithPlaintextFormat sets the log format to plaintext
+func WithPlaintextFormat() Option {
+	return func(spec *corev1.PodSpec) {
+		spec.Containers[0].Args = append(spec.Containers[0].Args, "--format")
+		spec.Containers[0].Args = append(spec.Containers[0].Args, "plaintext")
+	}
+}
+
+// WithText sets a custom text to be logged in each plaintext log
+func WithText(text string) Option {
+	return func(spec *corev1.PodSpec) {
+		spec.Containers[0].Args = append(spec.Containers[0].Args, "--text")
+		spec.Containers[0].Args = append(spec.Containers[0].Args, text)
+	}
+}
+
+func NewDeployment(namespace string, opts ...Option) *kitk8s.Deployment {
+	return kitk8s.NewDeployment(DefaultName, namespace).WithPodSpec(PodSpec(opts...))
+}
+
+func PodSpec(opts ...Option) corev1.PodSpec {
+	spec := corev1.PodSpec{
+
+		Containers: []corev1.Container{
+			{
+				Name:  DefaultContainerName,
+				Image: DefaultImageName,
+				Args: []string{
+					"--rate",
+					"10",
+				},
+				ImagePullPolicy: corev1.PullAlways,
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("100m"),
+						corev1.ResourceMemory: resource.MustParse("64Mi"),
+					},
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("1100m"),
+						corev1.ResourceMemory: resource.MustParse("128Mi"),
+					},
```

Comment:
based on the latest tests with the updated generator we can decrease the requests and limits